src/EditorFeatures/CSharpTest/Diagnostics/UseAutoProperty/UseAutoPropertyTests.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.CSharp.UseAutoProperty;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.UseAutoProperty
+{
+    public class UseAutoPropertyTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override Tuple<DiagnosticAnalyzer, CodeFixProvider> CreateDiagnosticProviderAndFixer(Workspace workspace)
+        {
+            return Tuple.Create<DiagnosticAnalyzer, CodeFixProvider>(
+                new UseAutoPropertyAnalyzer(), new UseAutoPropertyCodeFixProvider());
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestSingleGetter1()
+        {
+            Test(
+@"class Class { [|int i|]; int P { get { return i; } } }",
+@"class Class { int P { get; } }");
```

Comment:
This is a good point.  I will fix things up.  I'm going to have to talk to @jasonmalinowski  about this a bit though.  We'll be doing a FAR to determine if there are any writes outside of a constructor.  Then we're going to be doing a rename.  It seems like rename could use some refactoring to allow it to either use or expose the results of FAR so that you don't have to perform the operation twice.